.github/workflows/terraform-plan-apply.yml

```diff
@@ -333,11 +338,35 @@ jobs:
           az storage blob delete-batch --connection-string ${ui_storage_conn_string} --source "\$web" --pattern "*"
           az storage blob upload-batch --connection-string ${ui_storage_conn_string} --source mts-trial-ui --destination '$web'
 
+      # Download init service progress log file from azure file share and validate success
+      - name: Validate Init service progress
+        timeout-minutes: 30
+        # if: github.ref == 'refs/heads/main'
+        run: |
+          set -o nounset
+          #set -o xtrace
+          # can't use errexit and pipefail flags here becasue tail will fail if the file doesn't exists
+
+          echo "trying to fetch init service log in loop, timeout set to 30 min."
+          while [ true ]
+          do
+            sleep 10
+            az storage file download --no-progress --path log.txt --share-name ${init_storage_share_name} --connection-string ${init_storage_conn_string} && cat log.txt
+            result_failure=$(tail log.txt --lines 1 | grep "***FAILURE***" | wc -w)
+            result_success=$(tail log.txt --lines 1 | grep "***SUCCESS***" | wc -w)
+            if [ $result_failure -gt 0 ]; then
+                echo "Init service - Completed with failure"
+                exit 1
+            elif [ $result_success -gt 0 ]; then
+                echo "Init service - Completed Successfully"
+                exit 0
+            fi
```

Comment:
Is it really important to look at the last line?
If not, you could do something like: grep -q "some string" log.txt
and then check if return 0 (meaning found) or not.

Reply:
do you think there will be a scenario when we run the init process twice without deleting the log file?
incase things go wrong, will the developer delete the resource group and redeploy it with a new name?

Reply:
ok, so leave it with the tail.
anyway, you can simplify it by doing something like
if [ $(tail log.txt --lines 1 | grep -q "***FAILURE***") -eq 0 ]; then

Reply:
I've implemented it like this to avoid "unary operator expected" error and so git workflow won't fail, I left a comment in the code:
```
            tail log.txt --lines 1 | grep -q "FAILURE" && :
            if [ $? -eq 0 ]
            then
                echo "Init service - Completed with failure"
                exit 1
            fi

            tail log.txt --lines 1 | grep -q "SUCCESS" && :
            if [ $? -eq 0 ]
            then
                echo "Init service - Completed Successfully"
                exit 0
            fi
```